app/controllers/menu_controller.rb

```diff
@@ -31,52 +37,58 @@ def find_dishes(meal_type,day)
         return error, dishes
     end 
 
+
+    # GET 
+    # displays the menu, accepts optional query parameter date 
+    # if date is provided, shows menu for that date, otherwise shows 
+    # menu for today.
+    # Renders views/menu/menu.html
     def menu
-        # 1'st day hardcoded for now that the db and seed doesn't have all the data. Later, well use date
         date = params[:date] || Time.now.strftime("%d/%m/%Y")
         error, @dinner_dishes = find_dishes("Dinner",date)
         error, @supper_dishes = find_dishes("Supper",date)
         flash[:error] = error
     end 
 
 
+    # GET
     # edit the menu (add/remove dish)
-    # week, day are url params (see routes)
-    # request.body requires: 
-    #   - change_action: add OR remove
-    #   - dish_name: name of the dish
+    # url parameters required: 
+    #   - date 
+    #   - action: add OR remove
+    #   - dish: name of the dish
     #   - type: type of the meal
+    #   _ permanent: true or false -> Is the edit permanent or temporary
     # if things go wrong, spits out a flash message
    def edit
-    week = params[:week].to_i
-    day = params[:day].to_i
-    error = nil
-    if week > 6 || week < 0 || day > 6 || day < 0
-        error = 'Impossible week/day url parameters'
-    else 
-        begin
-            action = params[:change_action]
-            dish_name = params[:dish_name]
-            type = params[:type]
-            dish = Dish.get_dish(dish_name)
-            day = 7*(week) + day + 1
-            if action == nil || dish_name == nil
-                error = "Action or dish name is missing in request body"
-            end
+    begin
+        date = params[:date]
+        action = params[:act]
+        dish_name = params[:dish]
```

Comment:
How is this going to parse a dish name with spaces?

Reply:
rails handles that. I tried it with Rice Pilaf via ajax call, and as long as you encode it properly it works